server/chat-commands/room-sections.ts

```diff
@@ -0,0 +1,99 @@
+import {Utils} from '../../lib';
+
+export const sections = [
+	'none', 'nonpublic', 'officialrooms', 'officialtiers', 'communityprojects', 'gaming', 'languages', 'entertainment', 'lifehobbies', 'onsitegames',
+] as const;
+
+export type RoomSection = typeof sections[number];
+
+export const sectionNames: {[k in RoomSection]: string} = {
+	none: 'none',
+	nonpublic: 'Non-public',
+	officialrooms: 'Official rooms',
+	officialtiers: 'Official tiers',
+	communityprojects: 'Community projects',
+	gaming: 'Gaming',
+	languages: 'Languages',
+	entertainment: 'Entertainment',
+	lifehobbies: 'Life & hobbies',
+	onsitegames: 'On-site games',
+};
+
+export const commands: ChatCommands = {
+	setroomsection: 'roomsection',
+	roomsection(target, room, user) {
+		room = this.requireRoom();
+		if (!target) {
+			if (!this.runBroadcast()) return;
+			this.sendReplyBox(Utils.html`This room is ${sectionNames[room.settings.section] !== 'none' ? `in the ${sectionNames[room.settings.section]} section` : `not in a section`}.`);
+			return;
+		}
+		this.checkCan('gdeclare');
+		const section = room.adjustSection(target);
+		this.sendReply(`The room section is now: ${sectionNames[section]}`);
+
+		this.privateModAction(`(${user.name} changed the room section to: "${sectionNames[section]}".)`);
+		this.modlog('ROOMSECTION', null, `to "${sectionNames[section]}"`);
+	},
+	roomsectionhelp: [
+		`/roomsection [section] - Sets the room this is used in to the specified [section]. Requires: &`,
+		`Valid sections: ${sections.join(', ')}`,
+	],
+
+	desectionleader: 'sectionleader',
+	sectionleader(target, room, user, connection, cmd) {
+		this.checkCan('gdeclare');
+		room = this.requireRoom();
+		if (!target || target.split(',').length < 2) return this.parse(`/help sectionleader`);
+
+		const [targetStr, sectionid] = this.splitOne(target);
+		this.splitTarget(targetStr);
+		const section = room.sanitizeSection(sectionid);
+		const targetUser = this.targetUser;
+		const userid = toID(this.targetUsername);
+		const name = targetUser ? targetUser.name : this.targetUsername;
+		const demoting = cmd === 'desectionleader';
+		const leadsSection = Users.globalAuth.sectionLeaders.getSection(targetUser || userid);
+		if (Users.globalAuth.sectionLeaders.has(targetUser || userid) && !demoting && section === leadsSection) {
```

Comment:
```suggestion
		if (Users.globalAuth.sectionLeaders.has(targetUser || userid) && !demoting) {
```
Your code assumes people will only lead one section at a time, and your help message says that they should be demoted from Section Leader first. I spoke to US and we're not planning to have people lead two sections at once. (I'm fine with having `/sectionleader` change the section they lead, but then the help should be changed.)